.github/workflows/sanity-tests.yml

```
@@ -0,0 +1,48 @@
+name: Sanity tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  # Run once per week (Monday at 04:00 UTC)
+  schedule:
+    - cron: '0 4 * * 1'
+
+jobs:
+  build:
+    name: '${{ matrix.name }}'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: latest Ansible 5 release
+            ansible_major: 5
+            ansible_core: 2.12
+            ansible_core_next: 2.13
+          - name: latest Ansible 6 release
+            ansible_major: 6
+            ansible_core: 2.13
+            ansible_core_next: 2.14
+
+    steps:
+      - name: Check out ansible-build-data
+        uses: actions/checkout@v3
+        with:
+          path: ansible-build-data
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install ansible-core ${{ matrix.ansible_core }}
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install ansible-core>=${{ matrix.ansible_core }},${{ matrix.ansible_core_next }}
```

This way it should also install pre-releases of ansible-core, without installing pre-releases of ansible-core's dependencies:
```suggestion
          python3 -m pip install ansible-core>=${{ matrix.ansible_core }}a1,${{ matrix.ansible_core_next }}
```

---

I'm curious and want to test it out of curiosity, let me get back to you on that.